pkg/services/k8s/authnz/service.go

```diff
@@ -0,0 +1,141 @@
+package authnz
+
+import (
+	"context"
+	"fmt"
+
+	"net/http"
+	"strconv"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models/roletype"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/apikey"
+	"github.com/grafana/grafana/pkg/services/authn"
+	"github.com/grafana/grafana/pkg/services/authn/clients"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/web"
+	authnV1 "k8s.io/api/authentication/v1"
+	authzV1 "k8s.io/api/authorization/v1"
+)
+
+const GrafanaAdminK8sUser = "gl-admin"
```

Comment:
fine for now... but I think we can remove this entirely and use the grafana name

Reply:
Yeah once we can determine a stable name coming in from Grafana. At the moment, the stable names we have got are in the roles column (Admin, Editor, Viewer). The service accounts themselves have user-generated IDs, so they aren't stable.

<img width="1131" alt="Screenshot 2023-03-29 at 11 51 27 AM" src="https://user-images.githubusercontent.com/542168/228639093-19f80704-b0d3-4a1c-a15f-07c2265efeb2.png">